pages/configuration/high-availability.mdx

```diff
@@ -0,0 +1,264 @@
+until ping from coordinator passes. When the coordinator realizes that the once-MAIN instance is alive, it will choose between enabling writing on old MAIN
+or demoting it to REPLICA. The choice depends on whether there was any other MAIN instance chosen between old MAIN's failure and restart.
+
+Since instance can die and get back up, use `--replication-state-at-startup=true` to restore last replication state. With this config we will restore instance in 
+state it was last and everything should function normally from that point. In case this flag is not set, every instance will restart as MAIN and that case is not handled.
+
+## Handling errors
+
+Distributed system can fail in various ways. The logic is implemented in a way that Memgraph instance will be resilient to occasional network
+failures and independent machine failures. That's why there are parameters controlling frequency of health checks from coordinator to replication
+instances and the time needed to realize the instance is down.
+
+## Example
+
+This example will show how to setup highly-available cluster in Memgraph using a single coordinator.
+
+### 1. Start all instances
+
+i.) Start coordinator:
+```plaintext
+docker run  --name coord1 --rm -p 7687:7687 -p 7444:7444 memgraph/memgraph-mage --bolt-port=7687 --log-level=TRACE --data-directory=/tmp/mg_data_coord1 --log-file=/tmp/coord1.log --also-log-to-stderr --raft-server-id=1 --raft-server-port=10111 --experimental-enabled=high-availability
+```
+
+ii.) Start instance1:
+```plaintext
+docker run  --name instance1 --rm -p 7688:7688 -p 7445:7444 memgraph/memgraph-mage --bolt-port=7688 --log-level=TRACE --data-directory=/tmp/mg_data_instance1 --log-file=/tmp/instance1.log --also-log-to-stderr --replication-restore-state-on-startup --coordinator-server-port=10011 --experimental-enabled=high-availability
+```
+
+iii.) Start instance2:
+```plaintext
+docker run --name instance1 --rm -p 7689:7689 -p 7446:7444 memgraph/memgraph-mage --bolt-port=7689 --log-level=TRACE --data-directory=/tmp/mg_data_instance2 --log-file=/tmp/instance2.log --also-log-to-stderr --replication-restore-state-on-startup --coordinator-server-port=10012 --experimental-enabled=high-availability
+```
+
+iv.) Start instance3:
+```plaintext
+docker run --name instance1 --rm -p 7690:7690  -p 7447:7444 memgraph/memgraph-mage --bolt-port=7690 --log-level=TRACE --data-directory=/tmp/mg_data_instance3 --log-file=/tmp/instance3.log --also-log-to-stderr --replication-restore-state-on-startup --coordinator-server-port=10013 --experimental-enabled=high-availability
+```
+
+### 2. Register instances
+
+i.) Start communication with the coordinator with any Memgraph client on port 7687.
+```plaintext
+mgconsole --port=7687
+```
+ii.) Register all 3 instances as part of the cluster:
+```plaintext
+REGISTER INSTANCE instance_1 ON '127.0.0.1:10011' WITH '127.0.0.1:10001';
```

Comment:
If you are using a docker container, the IP should be the IP of the container since instances are not on the local host.

Reply:
Added callout "Replace `<ip_address>` with the container's IP address. This is necessary for Docker deployments where instances are not on the local host."